cmd/server-main.go

```diff
@@ -261,48 +260,36 @@ func isDistributedSetup(eps []*url.URL) bool {
 	return false
 }
 
-// We just exit for invalid endpoints.
-func checkEndpointsSyntax(eps []*url.URL, disks []string) {
+// Check if the endpoints are following expected syntax, i.e
+// valid scheme, valid path across all platforms.
+func checkEndpointsSyntax(eps []*url.URL, disks []string) error {
 	for i, u := range eps {
 		switch u.Scheme {
-		case "", "http", "https":
-			// Scheme is "" for FS and singlenode-XL, hence pass.
+		case "":
+			// "/" is not allowed.
+			if u.Path == "" || u.Path == "/" {
+				return fmt.Errorf("Root path is not allowed : %s (%s)", u.Path, disks[i])
+			}
+		case "http", "https":
+			// "http://server1/" is not allowed
+			if u.Path == "" || u.Path == "/" || u.Path == "\\" {
+				return fmt.Errorf("Root path is not allowed : %s (%s)", u.Path, disks[i])
+			}
 		default:
 			if runtime.GOOS == "windows" {
 				// On windows for "C:\export" scheme will be "C"
 				matched, err := regexp.MatchString("^[a-zA-Z]$", u.Scheme)
```

Comment:
You could compile the regex only once and reuse it everywhere
```
windowsSchemeRegex := regexp.MustCompile("^[a-zA-Z]$")
```

later

```
windowsSchemeRegex.MatchString(u.Scheme)
```

Another thought. As we use single alphabet check, is it possible to avoid regex?